src/js/messaging/containers/Main.jsx

```diff
@@ -18,13 +19,21 @@ class Main extends React.Component {
   constructor(props) {
     super(props);
     this.handleOnFolderChange = this.handleOnFolderChange.bind(this);
+    this.handleOnFolderNameChange = this.handleOnFolderNameChange.bind(this);
+    this.handleOnSubmit = this.handleOnSubmit.bind(this);
   }
 
   handleOnFolderChange(domEvent) {
     const folderId = domEvent.target.dataset.folderid;
     this.props.setCurrentFolder(folderId);
   }
 
+  handleOnFolderNameChange(field) {
```

Comment:
I typically see `handle*` and `on*` as complementary for naming event handling, but not together. Maybe simplify these to just `handle*`. For this one in particular, `handleNewFolderNameChange` to match the suggested change of action name to `setNewFolderName`.